crates/fs/private/previous.rs

```diff
@@ -442,16 +442,47 @@ mod private_history_tests {
     use chrono::Utc;
     use proptest::test_runner::{RngAlgorithm, TestRng};
 
+    struct TestSetup {
+        rng: TestRng,
+        store: MemoryBlockStore,
+        hamt: Rc<PrivateForest>,
+        root_dir: Rc<PrivateDirectory>,
+        discrepancy_budget: usize,
+    }
+
+    impl TestSetup {
+        fn new() -> Self {
+            let mut rng = TestRng::deterministic_rng(RngAlgorithm::ChaCha);
+            let store = MemoryBlockStore::default();
+            let root_dir = Rc::new(PrivateDirectory::new(
+                Namefilter::default(),
+                Utc::now(),
+                &mut rng,
+            ));
+
+            Self {
+                rng,
+                store,
+                hamt: Rc::new(PrivateForest::new()),
+                root_dir,
+                discrepancy_budget: 1_000_000,
+            }
+        }
+    }
+
     #[async_std::test]
     async fn previous_of_root_node() {
-        let rng = &mut TestRng::deterministic_rng(RngAlgorithm::ChaCha);
-        let store = &mut MemoryBlockStore::default();
-        let hamt = Rc::new(PrivateForest::new());
-        let root_dir = Rc::new(PrivateDirectory::new(
-            Namefilter::default(),
-            Utc::now(),
-            rng,
-        ));
+        let TestSetup {
+            mut rng,
+            mut store,
+            hamt,
+            root_dir,
+            discrepancy_budget,
+        } = TestSetup::new();
+
+        let rng = &mut rng;
```

Comment:
minor: can probably do w/o the lets here and just pass in the ref must into where they're used no? but no biggie.

Reply:
So far we went with `rng: &mut TestRng` in all other tests. I want to stay consistent with the other tests.

Reply:
@matheus23 oh I know, I just meant you can pass it in directly no instead of let vars?

Reply:
Well, I was trying to say "we're using a `let rng = &mut ...` in other tests, too, and I wanted to have that be consistent.

Reply:
@matheus23 ah, gotcha. Sorry for the confusion :).

Reply:
@matheus23 ah, gotcha. Sorry for the confusion :).